chmodr.js

```diff
@@ -5,54 +5,43 @@ var fs = require("fs")
 , path = require("path")
 
 function chmodr (p, mode, cb) {
-  fs.readdir(p, function (er, children) {
-    // any error other than ENOTDIR means it's not readable, or
-    // doesn't exist.  give up.
-    if (er && er.code !== "ENOTDIR")
-      return cb(er)
-    var isDir = !er
-    var m = isDir ? dirMode(mode) : mode
-    if (er || !children.length)
-      return fs.chmod(p, m, cb)
-
-    var len = children.length
-    var errState = null
-    children.forEach(function (child) {
-      var pathChild = path.resolve(p, child);
-      fs.lstat(pathChild, function(er, stats) {
-        if (er)
-          return cb(er)
-        if (!stats.isSymbolicLink())
-          chmodr(pathChild, mode, then)
-        else
-          then()
+  fs.lstat(p, function (er, stats) {
+    if (er) return cb(er)
+    if (stats.isSymbolicLink()) return cb()
+    if (stats.isDirectory()) {
+      fs.readdir(p, function (er, children) {
+        if (er) return cb(er)
+        if (!children.length) return fs.chmod(p, dirMode(mode), cb)
+        var len = children.length
+        var errState = null
+        children.forEach(function (child) {
+          chmodr(path.resolve(p, child), mode, then)
+        })
+        function then (er) {
+          if (errState) return
+          if (er) return cb(errState = er)
+          if (-- len === 0) return fs.chmod(p, dirMode(mode), cb)
+        }
       })
-    })
-    function then (er) {
-      if (errState) return
-      if (er) return cb(errState = er)
-      if (-- len === 0) return fs.chmod(p, dirMode(mode), cb)
+    } else {
+      return fs.chmod(p, mode, cb)
     }
   })
 }
 
 function chmodrSync (p, mode) {
   var children
-  try {
-    children = fs.readdirSync(p)
-  } catch (er) {
-    if (er && er.code === "ENOTDIR") return fs.chmodSync(p, mode)
-    throw er
+  var stats = fs.lstatSync(p)
+  if (stats.isSymbolicLink())
+    return;
```

Comment:
I think it'd be better to either use braces or make it a one-liner to keep style consistent with the rest of the file. Not sure if important, but figured I'd point it out (':